lib/sync.js

```diff
@@ -6,155 +6,96 @@ var _ = require('lodash'),
     i18nStrings = require('i18n-strings-files'),
     GS = require('google-spreadsheet'),
     stringFileLocator = require(path.resolve(__dirname, 'utils/stringFileLocator')),
-    updateStrings = require(path.resolve(__dirname, 'utils', 'updateStrings'));
+    updateStrings = require(path.resolve(__dirname, 'utils', 'updateStrings')),
+    Promise = require('bluebird'),
+
+    google = require(path.resolve(__dirname, 'providers', 'googleSpreadsheet')),
+    stringsUrl = require(path.resolve(__dirname, 'providers', 'stringsUrl'));
 
 var doubleEscape = function(string) {
     return string.replace(/[^\\]\\(n|r|t)/g, '\\\\$1');
 };
 
+var createProvider = function (config) {
```

Comment:
you can also make this more generic ... and use the filename as provider name ... then you don't need the switch here